packages/renderer/src/lib/deployments/DeploymentsList.spec.ts

```diff
@@ -154,3 +160,45 @@ test('Expect filter empty screen', async () => {
   const filterButton = screen.getByRole('button', { name: 'Clear filter' });
   expect(filterButton).toBeInTheDocument();
 });
+
+test('Expect user confirmation to pop up when preferences require', async () => {
+  const deployment: V1Deployment = {
+    apiVersion: 'apps/v1',
+    kind: 'Deployment',
+    metadata: {
+      name: 'my-deployment',
+      namespace: 'test-namespace',
+    },
+    spec: {
+      replicas: 2,
+      selector: {},
+      template: {},
+    },
+  };
+  kubernetesRegisterGetCurrentContextResourcesMock.mockResolvedValue([deployment]);
+
+  while (get(kubernetesCurrentContextDeployments).length === 0) {
+    await new Promise(resolve => setTimeout(resolve, 500));
+  }
```

Comment:
Let's avoid timeouts, and do something more `vitest` friendly.

I would recommend looking at https://vitest.dev/api/vi.html#vi-waitfor

I know this is some code that is already inside those test, but I believe this should be avoided 

```suggestion
  await vi.waitFor(() => get(kubernetesCurrentContextDeployments).length > 0);
```

Reply:
I changed it only in some of my tests, because in a few of them, this proposed change caused some problems with the rendering of components (not enough time?) leading to errors like `Unable to find an accessible element with the role...`